File: src/main/java/com/github/zoewithabang/model/Option.java
package com.github.zoewithabang.model;

import java.util.Objects;

public class Option implements IModel
{
    private String key;
    private String value;
    
    public Option() {}
    
    public Option(String key, String value)
    {
        this.key = key;
        this.value = value;
    }
    
    public String getKey()
    {
        return key;
    }
    
    public void setKey(String key)
    {
        this.key = key;
    }
    
    public String getValue()
    {
        return value;
    }
    
    public void setValue(String value)
    {
        this.value = value;
    }
    
    @Override
    public String toString()
    {
        return "Option{" +
            "key='" + key + '\'' +
            ", value='" + value + '\'' +
            '}';
    }
    
    @Override
    public boolean equals(Object o)
    {
        if(this == o) return true;
        if(o == null || getClass() != o.getClass()) return false;
        Option option = (Option) o;
        return Objects.equals(key, option.key) &&
            Objects.equals(value, option.value);
    }
    
    @Override
    public int hashCode()
    {
        return Objects.hash(key, value);
    }
}

File: src/main/kotlin/com/github/zoewithabang/model/Option.kt
package com.github.zoewithabang.model

class Option(key: String, value: String) {
    var key: String? = key
    var value: String? = value

    override fun toString(): String {
        return "Option(key=$key, value=$value)"
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Option

        if (key != other.key) return false
        if (value != other.value) return false

        return true
    }

    override fun hashCode(): Int {
        var result = key?.hashCode() ?: 0
        result = 31 * result + (value?.hashCode() ?: 0)
        return result
    }
}
